pkg/logbridge/logbridge_test.go

```diff
@@ -0,0 +1,50 @@
+package logbridge
+
+import (
+	"bufio"
+	"bytes"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func TestLogBridge(t *testing.T) {
+	type testCase struct {
+		inputSize      int
+		bridgeCapacity int
+		expected       int
+	}
+
+	testCases := []testCase{
+		testCase{inputSize: 0, bridgeCapacity: 0, expected: 0},
+		testCase{inputSize: 10, bridgeCapacity: 10, expected: 10},
+		testCase{inputSize: 100, bridgeCapacity: 10, expected: 10},
+		testCase{inputSize: 10, bridgeCapacity: 100, expected: 10},
+	}
+
+	for i, testCase := range testCases {
+		r := rand.New(rand.NewSource(time.Now().Unix()))
+		input := make([]byte, 0, 2*testCase.inputSize)
+		for i := 0; i < testCase.inputSize; i++ {
+			input = append(input, byte(r.Intn(256)), byte(10))
+		}
+
+		reader := bytes.NewReader(input)
+		buf := new(bytes.Buffer)
+		writer := bufio.NewWriter(buf)
```

Comment:
What's the bufio.Writer for? bytes.Buffer is already a writer; my instinctive impression (not saying it's correct, but this is what I think on first reading) is that the bufio wrapper is unnecessary. Does it impact test behavior?